model/src/main/scala/explore/model/ObsSummary.scala

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.model
+
+import cats._
+import explore.model.enum.ObsStatus
+import io.circe.Decoder
+import io.circe.Encoder
+import io.circe.HCursor
+import io.circe.Json
+import io.circe.syntax._
+import lucuma.core.model.Asterism
+import lucuma.core.model.Observation
+import lucuma.core.model.Target
+import monocle.macros.Lenses
+
+import java.time.Duration
+import java.time.temporal.ChronoUnit
+
+@Lenses
+final case class ObsSummary(
+  id:                Observation.Id,
+  name:              Option[String],
+  status:            ObsStatus = ObsStatus.New,
+  conf:              String = "GMOS-N R831 1x300",
+  constraints:       String = "<0.7\" <0.3 mag Bright",
+  duration:          Duration = Duration.of(93, ChronoUnit.MINUTES),
+  observationTarget: Option[Either[Asterism.Id, Target.Id]]
+)
+
+object ObsSummary {
+  implicit val eqObsSummary: Eq[ObsSummary] = Eq.by(x => (x.id, x.name, x.observationTarget))
+
+  implicit val obsSummaryEncoder: Encoder[ObsSummary] = new Encoder[ObsSummary] {
+    final def apply(c: ObsSummary): Json = {
+      val common = Json.obj(("id", c.id.asJson), ("name", c.name.asJson))
+      c.observationTarget match {
+        case None             => common
+        case Some(Right(tid)) =>
+          common.deepMerge(Json.obj(("observationTarget", Json.obj(("target_id", tid.asJson)))))
+        case Some(Left(aid))  =>
+          common.deepMerge(Json.obj(("observationTarget", Json.obj(("asterism_id", aid.asJson)))))
+      }
+    }
+
+  }
+
+  implicit val obsTargetDecoder: Decoder[Either[Asterism.Id, Target.Id]] =
+    new Decoder[Either[Asterism.Id, Target.Id]] {
+      final def apply(c: HCursor): Decoder.Result[Either[Asterism.Id, Target.Id]] =
+        for {
+          aid <- c.downField("asterism_id").as[Option[Asterism.Id]]
+          tid <- c.downField("target_id").as[Option[Target.Id]]
+        } yield (aid, tid) match {
+          case (Some(aid), _)    => Left(aid)
```

Comment:
Do we want to make `None` explicit in the 2nd position in this case? Otherwise, if both are defined it will match here instead of producing an error.